### PR TITLE
build(docker): optimize docker image layers

### PR DESCRIPTION
**Pull request by id**
`chown` only mutable state directories

Release version: v/e5.6.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Comment by zmstone** [comment]
maybe the executables, beams etc are better to be owned by root

**Review by thalesmg — DISMISSED** [review]

File: .github/workflows/build_and_push_docker_images.yaml [hunk 1]
```diff
@@ -204,6 +204,7 @@ jobs:
           docker exec -t -u root $CID node_dump
           docker rm -f $CID
       - name: Push docker image
+        if: inputs.publish || github.repository_owner != 'emqx'
```
[the remark this answers is not in the document]
Comment (id) [reply .github/workflows/build_and_push_docker_images.yaml:207]: No, the current condition is correct.
Was requested by William to be able to push from forks to custom registry.

**Review by zmstone — DISMISSED** [review]

**Comment by id** [comment]
oopsie

**Comment by zmstone** [comment]
right, there is no default `log` dir in docker image

**Review by zmstone — APPROVED** [review]

**Review by Rory-Z — APPROVED** [review]

**Comment by yanzhiemq** [comment]
image

After applied this optimization, EQMX image size has been decreased from 229MB to 132MB.

**Pull request merged**